mappings/clacso.py

```diff
@@ -1,23 +1,119 @@
+    def __init__(self, source, namespace, constants):
+        super(CLACSOMapping, self).__init__(source, namespace, constants)
         self.mapping = self.createMapping()
     
     def createMapping(self):
         return {
-            'title': ('title', '{0}'),
-            'authors': ('authors', '{0}|||true'),
-            'abstract': ('description', '{0}'),
-            'dates': [('publication_date', '{0}|publication_date')],
-            'identifiers': 
-                [('isbn', '{0}|isbn')]
-            ,
-            'spatial': ('edition_location', '{0}'),
-            'has_part': [('url', '1|{0}|clacso|application/pdf|{{"catalog": false, "download": true, "reader": false, "embed": false}}')] 
+            'title': ('./dc:title/text()', '{0}'),
+            'authors': [('./dc:creator/text()', '{0}|||true')],
+            'abstract': ('./dc:description/text()', '{0}'),
+            'dates': [
+                (
+                    [
+                        './dc:date/text()',
+                    ],
+                    '{0}'
+                )
+            ],
+            'identifiers': [
+                ('./dc:identifier/text()', '{0}'),
+            ],
+            'medium' : [
+                ('./dc:type/text()', '{0}')
+            ],
+            'has_part': [(
+                './dc:identifier/text()',
+                '1|{0}|clacso|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'
+            )]
         }
 
     def applyFormatting(self):
         self.record.source = 'clacso'
-        self.record.source_id = f'{self.record.source}'
+        handle_url = "https://biblioteca-repositorio.clacso.edu.ar/handle/CLACSO"
+        self.formatting_source_id(handle_url)
+        self.formatting_medium()
+        self.formatting_dates()
+        self.formatting_identifiers()
+        self.formatting_has_part(handle_url)
+
+    def formatting_source_id(self, handle_url):
+        for identifier in self.record.identifiers:
+            if handle_url in identifier:
+                id = identifier.split('/')[-1]
+                self.record.source_id = f'{self.record.source}|{id}'
+                break
+
+    def formatting_medium(self):
+        if self.record.medium:
+            type_list = ['book', 'bookpart', 'part', 'chapter', 'bibliography', 'appendix', 'index',
+                    'foreword', 'afterword', 'bibliography', 'review', 'article', 'introduction']
+            for medium in self.record.medium:
+                if '/' in medium:
+                    medium_value = medium.split('/')[-1]
+                    if medium_value.lower() in type_list:
+                        self.record.medium = medium_value.lower()
+                        break
+                elif '/' not in medium and medium.lower() in type_list:
+                    medium_value = medium
+                    self.record.medium = medium_value.lower()
+                    break
+        else:
```

Comment:
Is this else clause necessary? I don't think we should include empty strings in the mapping

Reply:
The type of the medium is unicode and in the `createmapping()` method  I initialized it as a list since there can be multiple mediums in the DSpace xml file. I want to make sure the medium when it's None doesn't have an empty list as its value in the records table. For now I'll get rid of the else statement and see how the medium field values look in the database.